src/toxcore/dht/server/mod.rs

```diff
@@ -1378,6 +1394,11 @@ impl Server {
     pub fn set_net_crypto(&mut self, net_crypto: NetCrypto) {
         self.net_crypto = Some(net_crypto);
     }
+
+    /// Get `PrecomputedKey`s cache.
+    pub fn get_precomputed_keys(&self) -> PrecomputedCache {
```

Comment:
What's that function for?

Reply:
It will be used to get cache from the `Server` to use it in other modules like `net_crypto`. It's supposed to be shared. Otherwise it makes little sense to have cache in `net_crypto`.

Reply:
Alright. Show me usage in examples/dht.rs.

Reply:
There is another way it could be done - just pass this cache as arg to both `Server::new` and `NetCrypto::new`. But I decided that it can be created internally in the `Server` and taken later with this method. In this way `Server::new` has less args.

Reply:
Can we set the cache for net_crypto when we call DhtServer::set_net_crypto?

Reply:
In this case we will have to create it in `NetCrypto` and then drop it and replace with cache from `DhtServer` - it's a bit untidy.